demo/kubernetes/tpr.md

```diff
@@ -0,0 +1,32 @@
+# Rook TPRs
+Rook allows creation and customization of storage clusters through the third party resources (TPRs). The following settings are available
+for a cluster.
+
+## Sample
+```
+apiVersion: rook.io/v1beta1
+kind: Cluster
+metadata:
+  name: my-rook
+spec:
+  namespace: rook
+  version: latest
+  useAllDevices: false
+  deviceFilter: ^sd.
+```
+
+## Settings
+Settings can be specified at the global level to apply to the cluster as a whole, while other settings can be specified at more fine-grained levels.
+
+### Global settings
+- `namespace`: The Kubernetes namespace that will be created for the Rook cluster. The services, pods, and other resources created by the operator will be added to this namespace. Each cluster must have a unique namespace. The common scenario is to create a single Rook cluster. If multiple clusters are created, they must not have conflicting devices or host paths.
+- `version`: The version of the `rook/rookd` and `rook/rook-operator` containers that will be deployed. Upgrades are not yet supported if this setting is updated for an existing cluster, but upgrades will be coming.
+- `useAllDevices`: `true` or `false`, indicating whether all devices found on nodes in the cluster should be automatically consumed by OSDs. Not recommended unless you have a very controlled environment where you will not risk formatting of devices with existing data. When `true`, all devices will be used except those with partitions created or a local filesystem. Is overridden by `deviceFilter` if specified.
+- `deviceFilter`: A regular expression that allows selection of devices to be consumed by OSDs. If not specified, devices will not be selected. This directly uses the golang regular expression. See the syntax [here](https://golang.org/pkg/regexp/syntax/). For example:
+   - `sdb`: Only selects the `sdb` device if found on a node
+   - `^sd.`: Selects all devices starting with `sd`
+   - `^sd[a-d]`: Selects devices starting with `sda`, `sdb`, `sdc`, and `sdd` if found
+   - `^s`: Selects all devices that start with `s`
```

Comment:
a negative example would be useful here too, such as selects everything BUT devices that start with `rbd`